### PR TITLE
164826388 tertiili reports

### PR DESCRIPTION
**Pull request by rckopteri**
# Added
* Tertiili graph and csv report
   
# Fixed
* All admin reports

### CODE REVIEW

File: ote/src/cljs/ote/views/monitor.cljs [hunk 1]
```diff
@@ -55,29 +66,70 @@
...
+         [:p "Termi "
+          [:strong "Liikkumispalveluiden"] " tarjoajat tarkoittaa sekä NAPiin rekisteröityneitä "
+          [:strong " liikkumispalveluiden tuottajia"] ", että sellaisia yrityksiä, joista on maininta NAP:ssa
+          julkaistuiden palveluiden tiedoissa (ns. " [:strong "palveluiden tuottamiseen osallistuvat yritykset"] ").
+          Esimerkki palvelun tuottamiseen osallistuvasta yrityksestä on yksityinen elinkeinon harjoittaja,
+          joka ajaa ainoastaan taksivälityskeskuksen kyytejä."
+          ]
+         [:h2 " Liikkumispalveluiden tarjoajien lukumäärän kehitys kuukausittain "]
+         [ui-common/linkify " /admin/reports/monitor/csv/monthly-companies " " Lataa kuvaajan tiedot CSV:nä "]
+         [:div {:style {:width " 75 % "}}
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/monitor.cljs]: Why `" 75 % "` instead of `"75%"`?

File: ote/src/cljs/ote/views/monitor.cljs [hunk 2]
```diff
@@ -55,29 +66,70 @@
...
+          [barchart-inner " bar-companies-by-month " nil companies-by-month-data]]
+
+         [:h2 " Liikkumispalveluiden tarjoajien lukumäärän kehitys tertiileittäin "]
+         [ui-common/linkify " /admin/reports/monitor/csv/tertiili-companies " " Lataa kuvaajan tiedot CSV:nä "]
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/monitor.cljs]: Just double checking that is linkify without url base ok?

File: ote/src/clj/ote/services/reports.sql [hunk 3]
```diff
@@ -161,35 +161,49 @@ END	AS "service-type",
...
+        ac."sub-type"
+   FROM
+        "all-companies" ac
+ GROUP BY month, ac."sub-type"
+ ORDER BY month, ac."sub-type";
+
+-- name: tertiili-producer-types-and-counts
+SELECT count(ac."sub-type") as sum,
+       ac."sub-type" as "sub-type",
+       concat(to_char(date_trunc('year', ac.created), 'YYYY'), ' ', (floor(extract(month FROM ac.created)::int / 4.00001)) + 1 , '/3 ') as tertiili
```
Comment (philosotractor) [inline ote/src/clj/ote/services/reports.sql]: Why `+ 1`?

File: ote/src/cljs/ote/views/monitor.cljs (hunk 1 above)
Comment (rckopteri) [reply ote/src/cljs/ote/views/monitor.cljs]: My bad. Thanx. Will fix

File: ote/src/cljs/ote/views/monitor.cljs (hunk 2 above)
Comment (rckopteri) [reply ote/src/cljs/ote/views/monitor.cljs]: yes, I think so

File: database/src/main/resources/db/migration/V1_146__all_companies_view.sql [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  SELECT null as "operator-id",
+         c.name as "business-name",
+         c."business-id" AS "business-id",
+         sc.created as created,
+         'service_company' as "source",
+         CASE "brokerage?" WHEN true THEN 'brokerage' ELSE "sub-type" END as "sub-type"
+   FROM
+        "group" g, "transport-operator" top
+        JOIN "transport-service" as ts ON ts."transport-operator-id" = top.id AND ts."published" IS NOT NULL AND ts.companies != '{"(,)"}',
+        service_company sc,
```
Comment (philosotractor) [inline database/src/main/resources/db/migration/V1_146__all_companies_view.sql]: This fetches all rows from `service_company`, should it?

Comment (rckopteri) [reply database/src/main/resources/db/migration/V1_146__all_companies_view.sql]: Really good catch. Fixed.

**Pull request merged**